app/controllers/api/v1/bookmarks_controller.rb

```diff
@@ -0,0 +1,117 @@
+class Api::V1::BookmarksController < Api::V1::BaseController
```

Comment:
As discussed previously, I've added some error handling. As a general rule, I like to delegate as much as possible to the back-end when writing an API, as the model should be performing essential data validation before anything goes into the database, and this theoretically ensures consistency between the UI and the API.

However, the ExternalWorks model doesn't check for the presence of a fandom and returns a weird message for a missing author, so I've intercepted some invalid requests in the API controller to return helpful messages from there instead, and added tests to check that this is happening.